src/test/app/PayChan_test.cpp

```diff
@@ -1136,6 +1152,101 @@ struct PayChan_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testMetaAndOwnership()
+    {
+        testcase("Metadata & Ownership");
+
+        using namespace jtx;
+        using namespace std::literals::chrono_literals;
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const settleDelay = 100s;
+        auto const pk = alice.pk();
+
+        auto inOwnerDir = [](ReadView const& view,
+                             Account const& acc,
+                             std::shared_ptr<SLE const> const& chan) -> bool {
+            ripple::Dir ownerDir(view, keylet::ownerDir(acc.id()));
+            return std::find(ownerDir.begin(), ownerDir.end(), chan) != ownerDir.end();
+        };
+
+        auto ownerDirCount = [](ReadView const& view,
+                                Account const& acc) -> std::size_t
+        {
+            ripple::Dir ownerDir(view, keylet::ownerDir(acc.id()));
+            return std::distance(ownerDir.begin(), ownerDir.end());
+
+        };
+
+        {
+            // Test without adding the paychan to the recipient's owner
+            // directory
+            Env env(
+                *this, supported_amendments() - fixPayChanRecipientOwnerDir);
+            env.fund (XRP (10000), alice, bob);
+            env(create(alice, bob, XRP(1000), settleDelay, pk));
+            auto const [chan, chanSle] = channelKeyAndSle(*env.current(), alice, bob);
+            BEAST_EXPECT(inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 1);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+            // close the channel
+            env(claim(bob, chan), txflags(tfClose));
+            BEAST_EXPECT(!channelExists(*env.current(), chan));
+            BEAST_EXPECT(!inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 0);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+        }
+
+        {
+            // Test with adding the paychan to the recipients owner directory
```

Comment:
nit: recipient's

Reply:
Fixed in 499e9e0e3 `Add `close` after channel create and fix typos`